POEM_052.md

```diff
@@ -242,23 +344,22 @@ def J_some_func(x, y, z, J):
     J['bar', 'x'][:] = 2 # need to set all elements of array
     J['bar', 'y'][:] = 1 # need to set all elements of array
 
+@om.func.declare_partials(of='foo', wrt='*', rows=np.arange(4), cols=np.arange(4))
+@om.func.declare_partials(of='bar', wrt=('x', 'y'), rows=np.arange(4), cols=np.arange(4))
```

Comment:
I really want to keep the most core API completely independent of OpenMDAO. that means that you have to be able to provide all information using annotations and basic python data structures (e.g. dict, tuple). 

We can provide helper classed like Metadata containers, and decorators to make slightly cleaner syntax. But since this API could make engineering code 100% independent of an explicit OpenMDAO dependency, I think we should try to allow that. It will be useful to any org who has some users that use OM and others that don't. The ones who don't will not want to install OM to be able to use their engineering code.